lib/models/consent.dart

```diff
@@ -0,0 +1,236 @@
+import 'package:json_annotation/json_annotation.dart';
+import 'package:pispapp/models/model.dart';
+import 'package:pispapp/models/party.dart';
+
+part 'consent.g.dart';
+
+@JsonSerializable(explicitToJson: true, includeIfNull: false)
+class Consent implements JsonModel {
+  Consent({
+    this.id,
+    this.consentId,
+    this.party,
+    this.status,
+    this.userId,
+    this.consentRequestId,
+    this.accounts,
+    this.authChannels,
+    this.authUri,
+    this.authToken,
+    this.initiatorId,
+    this.participantId,
+    this.scopes,
+    this.credential,
+  });
+
+  @override
+  factory Consent.fromJson(Map<String, dynamic> json) => _$ConsentFromJson(json);
+
+  /// Internal id that is used to identify the transaction.
+  String id;
+
+  /// Common ID between the PISP and FSP for the Consent object. This tells
+  /// DFSP and auth-service which consent allows the PISP to initiate
+  /// transaction.
+  String consentId;
+
+  /// Information about the party that is associated with the consent.
+  Party party;
+
+  /// Information about the current status of the consent.
+  ConsentStatus status;
+
+  /// User Id provided by app
+  String userId;
+
+  /// Id required to identify a specific consent request
+  String consentRequestId;
+
+  /// List of accounts that exist for a given user
+  List<Account> accounts;
+
+  /// List of channels available for a user to authenticate themselves with
+  List<TAuthChannel> authChannels;
```

Comment:
I'm actually not sure whether we should use the `T` prefix here or be consistent with the naming :|

Reply:
I just referenced the name from the auth-service stuff. Not sure what to do with this? What do you think?

Reply:
I think it would be better to be consistent by not using the prefix `T`?